echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/utils/NodeIdentity.java

```diff
@@ -25,21 +25,21 @@
 
 /** Provides an identity for each echo node. */
 public class NodeIdentity {
-  public static final String UNKNOWN_HOST = "UnknownHost";
+  private static final String UNKNOWN_HOST = "UnknownHost";
 
-  private String runtimeName;
-  private String hostName;
-  @Getter private String identity;
+  @Getter private final String identity;
 
   public NodeIdentity() {
```

Comment:
nobody ever calls the other constructor, so can you just make `"www.google.com"` and `80` be constants?

And then, instead of doing a lot of expensive work in the constructor, which is generally discouraged, can you make it `private NodeIdentity(String identity)` and use a `public static NodeIdentity create()` method that does the heavy lifting?

Reply:
Given that we will probably cherry pick this into a few previous branches, I would prefer to keep the change as small and understandable as possible. You're suggesting good changes, so lets fix the bug first and then improve the code health.

Reply:
Okay, fair enough.